content/en/blog/_posts/2020-12-10-Granular-Control-of-Volume-Permission-Changes.md

```diff
@@ -0,0 +1,55 @@
+--- 
+layout: blog 
+title: 'Kubernetes  1.20: Granular Control of Volume Permission Changes'
+date: 2020-12-10 
+slug: kubernetes-release-1.20-fsGroupChangePolicy-fsGroupPolicy
+---
+
+**Authors**: Hemant Kumar, Red Hat & Christian Huffman, Red Hat
+
+Kubernetes 1.20 brings two important beta features, allowing Kubernetes admins and users alike to have more adequate control over how volume permissions are applied when a volume is mounted inside a Pod.
+
+### Allow users to skip recursive permission changes on mount
+Traditionally if your pod is running as a non-root user ([which you should](https://twitter.com/thockin/status/1333892204490735617)), you must specify a fsGroup inside the pod’s security context so that the volume can be readable and writable by the Pod. This requirement is covered in more detail in [here](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/).
+
+But one side-effect of using fsGroup is that each time a volume is mounted, Kubernetes must recursively chown and chmod all the files inside the volume (with a few exceptions as noted below). This happens even if group ownership of the volume already matches the requested fsGroup, and can be pretty expensive for larger volumes with lots of small files while causing pod startup to take a long time. This scenario has been a known problem [for a while](https://github.com/kubernetes/kubernetes/issues/69699), and in Kubernetes 1.20 we are providing knobs to opt-out of recursive permission change if the volume already has the correct permissions.
+
+When configuring a pod’s security context, configure fsGroupChangePolicy with “OnRootMismatch” so if the root of the volume already has the correct permissions, the recursive permission change can be skipped. Kubernetes ensures that permissions of the top-level directory are changed last the first time it applies permissions. 
+
+```yaml
+securityContext:
+  runAsUser: 1000
+  runAsGroup: 3000
+  fsGroup: 2000
+  fsGroupChangePolicy: "OnRootMismatch"
+```
+More details about this can be found on - https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#configure-volume-permission-and-ownership-change-policy-for-pods 
```

Comment:
I think maybe wrapping the link in a text would be a lot cleaner here.

Reply:
Agreed, I'll fix this shortly, along with the other edits requested above.